### PR TITLE
feat: Update SchurComplement.lean by adding matrix determinant lemma

### PR DESCRIPTION
**Pull request by liuyf0188**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Matrix/SchurComplement.lean [hunk 1]
```diff
@@ -444,14 +444,20 @@ theorem det_one_sub_mul_comm (A : Matrix m n α) (B : Matrix n m α) :
...
-/-- A special case of the **Matrix determinant lemma** for when `A = I`.
-
-TODO: show this more generally. -/
+/-- A special case of the **Matrix determinant lemma** for when `A = I`.-/
 theorem det_one_add_col_mul_row (u v : m → α) : det (1 + col u * row v) = 1 + v ⬝ᵥ u := by
   rw [det_one_add_mul_comm, det_unique, Pi.add_apply, Pi.add_apply, Matrix.one_apply_eq,
     Matrix.row_mul_col_apply]
 #align matrix.det_one_add_col_mul_row Matrix.det_one_add_col_mul_row
 
+/-- general case of the **Matrix determinant lemma** for when `A = I`.-/
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Matrix/SchurComplement.lean:453]:
```suggestion
/-- The **Matrix determinant lemma** -/
```

File: Mathlib/LinearAlgebra/Matrix/SchurComplement.lean [hunk 2]
```diff
@@ -444,14 +444,20 @@ theorem det_one_sub_mul_comm (A : Matrix m n α) (B : Matrix n m α) :
...
+theorem matrix_determinant_lemma
+  (A: Matrix m m α)(U: Matrix m n α)(V: Matrix n m α)(hA: IsUnit A.det):
+  (A + U * V).det = A.det*(1 + V * (A⁻¹) * U).det := by
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Matrix/SchurComplement.lean:456]: Can you adjust the spacing and indentation here to match the rest of this file? Thanks!

**Review by eric-wieser — COMMENTED** [review]
This looks great, thanks! Just a few minor comments

File: Mathlib/LinearAlgebra/Matrix/SchurComplement.lean (hunk 1 above)
Comment (liuyf0188) [reply Mathlib/LinearAlgebra/Matrix/SchurComplement.lean]: sure, thanks for the comment

File: Mathlib/LinearAlgebra/Matrix/SchurComplement.lean [hunk 3]
```diff
@@ -450,10 +450,9 @@ theorem det_one_add_col_mul_row (u v : m → α) : det (1 + col u * row v) = 1 +
     Matrix.row_mul_col_apply]
 #align matrix.det_one_add_col_mul_row Matrix.det_one_add_col_mul_row
 
-/-- general case of the **Matrix determinant lemma** for when `A = I`.-/
-theorem matrix_determinant_lemma
-  (A: Matrix m m α)(U: Matrix m n α)(V: Matrix n m α)(hA: IsUnit A.det):
-  (A + U * V).det = A.det*(1 + V * (A⁻¹) * U).det := by
+/-- The **Matrix determinant lemma** -/
+theorem matrix_determinant_lemma (A: Matrix m m α) (U: Matrix m n α) (V: Matrix n m α) (hA: IsUnit A.det):
```
Comment (liuyf0188) [inline Mathlib/LinearAlgebra/Matrix/SchurComplement.lean:454]: i have adjusted the spacing and indentation, does it look good now?

**Comment by ADedecker** [comment]
@liuyf0188 Is this ready for review again? If so, please add the "awaiting-review" label, otherwise reviewers and maintainers might miss your PR!

**Comment by eric-wieser** [comment]
I've just noticed that unfortunately this is from a fork, so CI isn't running

**Comment by liuyf0188** [comment]
I think it is ready to be reviewed, please let me know what should i do.

**Comment by ADedecker** [comment]
Yes, you shouldn't do pull requests from forks, instead you should work on a mathlib branch directly. See https://leanprover-community.github.io/contribute/index.html for more information.

**Comment by liuyf0188** [comment]
ok thank you, I'll try create a different pull request

**Comment by liuyf0188** [comment]
i have created a new pull request [here[](https://github.com/leanprover-community/mathlib4/pull/7086), the ci seems can run but one of the test failed, can anyone help me under that pull request? and does it mean i can close this one? thanks.

**Comment by ADedecker** [comment]
Yes you can close this. Please mention it in the description of the other PR. I'll answer the CI question there.

**Comment by liuyf0188** [comment]
> Yes you can close this. Please mention it in the description of the other PR. I'll answer the CI question there.

Thank you

**Pull request closed without merging**